lib/cards/KCard.vue

```diff
@@ -273,31 +288,58 @@
       return {
         mouseDownTime: 0,
         ThumbnailDisplays,
+        isThumbnailImageLoaded: false,
+        isLinkFocused: false,
       };
     },
     computed: {
-      coreOutlineFocus() {
-        return {
-          ':focus': {
-            ...this.$coreOutline,
-          },
-        };
+      focusStyle() {
+        return this.isLinkFocused ? this.$coreOutline : {};
+      },
+      /**
+       * Disable the thumbnail placeholder element when
+       * there is no thumbnail area or the placeholder element
+       * is not provided.
+       *
+       * Furthermore, hide it after the thumbnail image
+       * is successfully loaded. Otherwise in some scenarios,
+       * such as when there is a large placeholder element
+       * and a small thumbnail image, some parts of the placeholder
+       * element may be visible behind the image after it has been
+       * successfully loaded.
+       *
+       * However, do not hide the placeholder element while
+       * the image is still loading to ensure progressive
+       * loading experience on slower networks.
+       */
+      disableThumbnailPlaceholder() {
+        return (
+          this.thumbnailDisplay === this.ThumbnailDisplays.NONE ||
+          !this.$slots.thumbnailPlaceholder ||
+          this.isThumbnailImageLoaded
+        );
+      },
+      hasAboveTitleArea() {
+        return this.$slots.aboveTitle || this.preserveAboveTitle;
+      },
+      hasBelowTitleArea() {
+        return this.$slots.belowTitle || this.preserveBelowTitle;
+      },
+      hasFooterArea() {
+        return this.$slots.footer || this.preserveFooter;
       },
       headingElement() {
         return this.headingLevel ? 'h' + this.headingLevel : undefined;
       },
-      layoutClass() {
-        return this.stylesAndClasses.layoutClass;
+      cardAreaClasses() {
```

Comment:
I think I'd need to hear a bit more to understand. `thumbnailStyles` are passed as `appearanceOverrides` prop to `KImg`, which is quite different from what `cardAreaClasses` do, so I'm not entirely sure how to connect these. If you had ideas on how to simplify, that's always nice - would you write some pseudo-code to demonstrate how you imagined the helper functions?

Reply:
I dont get it @AllanOXDi all of this is handled in the [stylesAndClasses](https://github.com/MisRob/kolibri-design-system/blob/3f42452154c8c0b72ecb66242c13aab024fc7583/lib/cards/KCard.vue#L353) computed prop, and there we already manage the logic for large/small/horizontal/vertical layouts, and I dont see duplicated code there, and yes, we do have some baseStyles for both of them: [thumbnailCommonStyles](https://github.com/MisRob/kolibri-design-system/blob/3f42452154c8c0b72ecb66242c13aab024fc7583/lib/cards/KCard.vue#L354), [cardAreaCommonClasses](https://github.com/MisRob/kolibri-design-system/blob/3f42452154c8c0b72ecb66242c13aab024fc7583/lib/cards/KCard.vue#L358).

Reply:
@AllanOXDi Let's look at

```
function getCardAreaClasses(orientation) {
// Return classes based on orientation
```

for example. This can't be achieved so easily, since `orientation` is not the only deciding factor (for one example, see horizontal orientation and larger thumbnail display). So in these new helpers, we'd basically need to start duplicating the orientation/thumbnailDisplay (=layout) conditions that are already in `stylesAndClasses`. Taking into account that main motivation for this particular organization was to avoid having the layout definitions scattered all over the place via many same conditions, I am afraid this strategy wouldn't help us overall. I can understand that it's a larger method and no problem to split it if you have other ideas, however it would need to happen in a way that would allow us to keep the single source of truth for layout behavior. Without it, it proved to be quite a pain point complicating any updates.